### PR TITLE
tests: Add code coverage to the repository

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,3 +2,5 @@ pytest<8.0
 filecheck<0.0.23
 lit<16.0.0
 frozenlist<1.4.*
+codecov
```
Comment (math-fehr) [inline requirements.txt:5]: Should this be in `requirements-optional.txt instead?

**Review by math-fehr — APPROVED** [review]
Besides the small nit, looks great!

**Review by webmiche — APPROVED** [review]
LGTM, thanks a lot, this has been necessary for ages! I am actually surprised that we get 81%, I feared we would have less. But I guess this might be the legendary 20/80 rule :wink:

File: .github/workflows/ci-core.yml [hunk 2]
```diff
@@ -32,10 +32,14 @@ jobs:
         pip install --upgrade pip
     - name: Install the package locally
       run: pip install -e .
-    - name: Test with pytest
+    - name: Test with pytest and check code coverage
       run: |
-        pytest tests/
+        pytest --cov --cov-config=.coveragerc --cov-report=xml tests/
     - name: Execute lit tests
```
[the remark this answers is not in the document]
Comment (georgebisbas) [reply .github/workflows/ci-core.yml:38]: Right, we need to look into alternatives I reckon, e.g. ? https://llvm.org/docs/CommandGuide/llvm-cov.html

File: requirements.txt (hunk 1 above)
Comment (georgebisbas) [reply requirements.txt:5]: Yes, moving.

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+codecov:
```
[the remark this answers is not in the document]
Comment (webmiche) [reply codecov.yml:1]: Ah yes, you are uploading the coverage as part of the testing CI job. Make sense :+1:

**Pull request merged**